### PR TITLE
Add policy document for using GitHub actions in workflows

### PR DESCRIPTION
**Pull request by bjhargrave**
Closes https://github.com/instructlab/dev-docs/issues/12

### CODE REVIEW

**Review by hickeyma — APPROVED** [review]
LGM, thanks @bjhargrave.
One thing, would it be worth discussing "secrets"?

**Review by nathan-weinberg — APPROVED** [review]
LGTM, thanks @bjhargrave!

**Comment by bjhargrave** [comment]
> One thing, would it be worth discussing "secrets"?

I think a doc on secret management for the github organization is also a good idea. There could be a list of secrets, their purposed, how they are managed, who is responsible for rotation if needed, etc.

File: docs/github-actions-use-policy.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  uses: step-security/harden-runner@a4aa98b93cab29d9b1101a6143fb8bce00e2eac4 # v2.7.1
+  with:
+    disable-sudo: true
+    egress-policy: block
+    allowed-endpoints: >
+      github.com:443
+```
+
+Hardening the action runner like this can prevent malicious or misbehaving actions or the misuse of actions from exfiltrating secrets.
+The article [Harden-Runner Defends Against Arbitrary Command Execution in tj-actions/changed-files GitHub Action](https://www.stepsecurity.io/blog/defend-against-arbitrary-command-execution-in-tj-actions-changed-files) shows how the misuse of an action could allow an attacker with a well-crafted pull request to exfiltrate secrets.
```
Comment (russellb) [inline docs/github-actions-use-policy.md:90]:
```suggestion
The article [Harden-Runner Defends Against Arbitrary Command Execution in `tj-actions/changed-files` GitHub Action](https://www.stepsecurity.io/blog/defend-against-arbitrary-command-execution-in-tj-actions-changed-files) shows how the misuse of an action could allow an attacker with a well-crafted pull request to exfiltrate secrets.
```

This would make it so `tj` doesn't need to be in spellcheck

File: docs/github-actions-use-policy.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+## Trusted Providers of GitHub Actions
+
+There are many GitHub Actions available in GitHub.
+Not all can be necessarily trusted.
+The InstructLab project must maintain a list of allowed providers and a list of denied providers.
+
+Allowed providers will include all GitHub created actions (`actions/*`) as well as other official actions such as Python Packaging actions (`pypa/*`).
+We will need to inventory the current use of actions within the project's repositories to build out the list of allowed and denied providers.
+
+The InstructLab organization's Settings->Actions->General must be configured to allow select actions including actions created by GitHub along with the allowed providers.
```
Comment (russellb) [inline docs/github-actions-use-policy.md:61]: Would you like to file an issue that tracks getting to this goal?  Changing the GitHub config for the whole org seems like the finish line.

File: docs/github-actions-use-policy.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+Adding additional actions to the allowed providers list will require approval by the organization maintainers along with updating the organization's settings.
```
Comment (russellb) [inline docs/github-actions-use-policy.md:62]: Maybe reword this as submitting a PR to the list of approved actions in the dev-docs repo to make the process explicit?

**Review by russellb — COMMENTED** [review]
I like this in general.  I have a couple of minor suggestions, including making sure we file something to track the follow-up work.

File: docs/github-actions-use-policy.md (hunk 2 above)
Comment (nathan-weinberg) [reply docs/github-actions-use-policy.md:61]: Community repo would probably be the place to do so

Comment (russellb) [reply docs/github-actions-use-policy.md:61]: I'd actually prefer `dev-docs` -- it's more a central technical governance task

File: docs/github-actions-use-policy.md (hunk 3 above)
Comment (bjhargrave) [reply docs/github-actions-use-policy.md]: Done. I added another md file to hold the provider lists and populated it with the most used actions I observed.

**Review by hickeyma — COMMENTED** [review]
@russellb Do you mins reviewing this again?

**Review by russellb — APPROVED** [review]
Thanks for your work on this!
